src/main/webapp/app/components/commentIcon/CommentIcon.tsx

```diff
@@ -0,0 +1,260 @@
+        </div>
+      </DefaultTooltip>
+    </div>
+  );
+});
+
+export interface ICommentBoxProps {
+  commentStore: CommentStore;
+  comments: Comment[];
+  openCommentId: string;
+  onCreateComment: (content: string) => void;
+  onResolveComment: (index: number) => void;
+  onUnresolveComment: (index: number) => void;
+  onDeleteComments: (indices: number[]) => void;
+}
+
+const CommentBox = observer((props: ICommentBoxProps) => {
+  const scrollContainerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const scrollPosition = props.commentStore.openCommentsScrollPosition;
+    if (scrollPosition && scrollContainerRef.current) {
+      scrollContainerRef.current.scrollTop = scrollPosition;
+    }
+  });
+
+  useEffect(() => {
+    return () => {
+      if (props.comments && props.commentStore.openCommentsId !== props.openCommentId) {
+        props.onDeleteComments(props.commentStore.commentIndiciesToDelete);
+        runInAction(() => props.commentStore.clearCommentsToDelete());
+      }
+    };
+  }, []);
+
+  const sortedCommentItems = props.comments
+    ?.map((comment, index) => {
+      return [
+        comment.date,
+        <CommentItem
+          commentStore={props.commentStore}
+          key={`${comment.date}-${comment.email}`}
+          comment={comment}
+          index={index}
+          onResolveComment={() => {
+            runInAction(() => props.commentStore.setOpenCommentsScrollPosition(scrollContainerRef.current.scrollTop));
+            props.onResolveComment(index);
+          }}
+          onUnresolveComment={() => {
+            runInAction(() => props.commentStore.setOpenCommentsScrollPosition(scrollContainerRef.current.scrollTop));
+            props.onUnresolveComment(index);
+          }}
+        />,
+      ];
+    })
+    .sort((comment1, comment2) => {
+      return parseInt(comment2[0] as string, 10) - parseInt(comment1[0] as string, 10);
+    })
+    .map(comment => comment[1]);
+
+  return (
+    <>
+      {sortedCommentItems && sortedCommentItems.length > 0 && (
+        <div ref={scrollContainerRef} style={{ maxHeight: '320px', overflow: 'auto' }}>
+          <ListGroup className="ml-3 mr-3 mb-3 mt-3" flush>
+            {sortedCommentItems}
+          </ListGroup>
+        </div>
+      )}
+      <CommentInput
+        onCreateComment={content => {
+          runInAction(() => props.commentStore.setOpenCommentsScrollPosition(0));
+          props.onCreateComment(content);
+        }}
+      />
+    </>
+  );
+});
+
+export interface ICommentInputProps {
+  onCreateComment: (content: string) => void;
+}
+
+function CommentInput(props: ICommentInputProps) {
+  const [commentInputValue, setCommentInputValue] = useState('');
+
+  function handleKeyPress(event: React.KeyboardEvent<HTMLInputElement>) {
+    if (commentInputValue !== '' && event.key === 'Enter') {
+      props.onCreateComment(commentInputValue);
+    }
+  }
```

Comment:
👍🏼  thanks for adding. I do expect this to be a default behavior for input group with a primary action button components. cc @calvinlu3 
Not need to fix in this PR, we will need to do it in the next feature where the input group needed.